commcare-cloud/commcare_cloud/commands/ansible/ansible_playbook.py

```diff
@@ -229,3 +235,198 @@ def run(self, args, unknown_args):
+                    tags.append("%s_%s" % (action, service))
+            if tags:
+                unknown_args.append('--tags=%s' % ','.join(tags),)
+        unknown_args.extend(['--extra-vars', "desired_state=%s desired_action=%s" % (state, action)])
+        # ToDo: use this with when in the playbook instead of tags
+        # currently its running riak even when just riak-cs is ran
+        # but skips riak when running for only stanchion
+        # as of now it looks like
+        # --extra-vars 'desired_services=['"'"'riak-cs'"'"', '"'"'stanchion'"'"']'
+        unknown_args.extend(['--extra-vars', "desired_services=%s" % run_for_services])
+        AnsiblePlaybook(self.parser).run(args, unknown_args)
+
+    def run_for_kafka(self, args, unknown_args):
+        tags = []
+        action = args.action
+        state = self.DESIRED_STATE_FOR_ACTION[action]
+        args.playbook = "service_playbooks/kafka.yml"
+        run_for_services = self.run_for_services('kafka', args)
+        if args.only:
+            # for options to act on certain services create tags
+            for service in run_for_services:
+                if service:
+                    tags.append("%s_%s" % (action, service))
+            if tags:
+                unknown_args.append('--tags=%s' % ','.join(tags),)
+        unknown_args.extend(['--extra-vars', "desired_state=%s desired_action=%s" % (state, action)])
+        # ToDo: use this with when in the playbook instead of tags
+        unknown_args.extend(['--extra-vars', "desired_services=%s" % run_for_services])
+        AnsiblePlaybook(self.parser).run(args, unknown_args)
+
+    def run_for_pg_standby(self, args, unknown_args):
+        self.run_ansible_module_for_service_group('pg_standby', args, unknown_args, inventory_group="pg_standby")
+
+    def ensure_permitted_only_options(self, service_group, args):
+        run_for_services = self.run_for_services(service_group, args)
+        for service in run_for_services:
+            assert service in self.SERVICES[service_group], \
+                ("%s not allowed. Please use from %s for --only option" %
+                 (service, self.SERVICES[service_group])
+                 )
+
+    def perform_action(self, service_group, args, unknown_args):
+        if service_group in ['proxy', 'redis', 'couchdb2', 'postgresql', 'rabbitmq']:
+            self.run_ansible_module_for_service_group(service_group, args, unknown_args)
+        elif service_group == "riakcs":
+            self.run_for_riakcs(args, unknown_args)
+        elif service_group == "stanchion":
+            if not args.only:
+                args.only = "stanchion"
+            self.run_for_riakcs(args, unknown_args)
+        elif service_group == "es":
+            self.run_for_es(args, unknown_args)
+        elif service_group == "kafka":
+            self.run_for_kafka(args, unknown_args)
+        elif service_group == "pg_standby":
+            self.run_for_pg_standby(args, unknown_args)
```

Comment:
Other than "es" (which uses a different implementation) these other "special" cases look really similar to the default case. What's preventing them from using the `run_ansible_module_for_service_group` method like the others?

Reply:
@dannyroberts 
You're right, they are similar. I can extract `run_ansible_playbook_for_service_group` for riakcs, kafka and stanchion. Can't reuse `run_ansible_module_for_service_group` since that runs a module.

Reply:
Ah, right, sounds like I stated the wrong function, but sounds like you understood what I was trying to get at. Glad you were able to simplify a bit!